common.rkt

```diff
@@ -162,7 +165,23 @@
   (reify initial-var st))
 
 (define (term<? u v)
-  (and (not (eqv? (car u) '=/=)) (eqv? (car v) '=/=)))
+  (cond
+    ((null? u) (not (null? v)))
+    ((null? v) #f)
+    ((not u) (not v))
+    ((not v) #f)
+    (u v)
+    (v #f)
```

Comment:
You should be comparing with `#t` directly.  `cond` will consider anything that isn't `#f` to be true.